flatten_json.py

```diff
@@ -104,17 +104,19 @@ def unflatten_list(flat_dict, separator='_'):
 
     def _convert_dict_to_list(object_, parent_object, parent_object_key):
         if isinstance(object_, dict):
+            keys = []
             try:
-                keys = [int(key) for key in sorted(object_) if
-                        not isinstance(object_[key], Iterable) or isinstance(object_[key], str)]
-                keys_len = len(keys)
-                if (sum(keys) == int(((keys_len - 1) * keys_len) / 2) and keys[0] == 0 and keys[-1] == keys_len - 1 and
-                        check_if_numbers_are_consecutive(keys)):
-                    parent_object[parent_object_key] = [object_[str(key)] for key in keys]
+                keys = [int(key) for key in object_]
+                keys.sort()
```

Comment:
I like this. You'll not be incurring the cost of sorting the list if a key cannot be converted to string. Also you are not incurring the cost of creating a new list whereas `sorted()` does:
http://stackoverflow.com/questions/22442378/what-is-the-difference-between-sortedlist-vs-list-sort-python